### PR TITLE
Add arm64 support

### PR DESCRIPTION
**Pull request by samuelallan72**
- add arm64 support to charmcraft.yaml
- remove dependency on bootstack-actions func.yaml
- run func tests (includes charm build) on arm64 and amd64

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Check workflow running linter, unit and functional tests
+name: Tests
```
Comment (samuelallan72) [inline .github/workflows/check.yaml:1]: Original name was too long to read which variant was running.

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -26,18 +26,49 @@ jobs:
...
-      juju-channel: ${{ matrix.juju-channel }}
-      nested-containers: false
-      provider: "lxd"
-      python-version: "3.10"
-      timeout-minutes: 120
+        juju-channel: ["3.4/stable"]
+        architecture:
+          - runs-on: [ubuntu-latest]  # github hosted runners are amd64
+            model-constraints: ""
+            # TODO: should we use these or the Ubuntu_ARM64_4C_16G_01 github-hosted runner?
```
Comment (samuelallan72) [inline .github/workflows/check.yaml:40]: This is pending internal discussion.

**Review by Pjack — APPROVED** [review]
LGTM, do we still need to discuss topics?

**Comment by samuelallan72** [comment]
> LGTM, do we still need to discuss any topics?

Thanks @Pjack .  The only topic remaining is which arm64 runners to use (self-hosted pool or the github hosted one).  That's something we can easily change later, so I wouldn't consider it a blocking issue.

File: .github/workflows/check.yaml (hunk 1 above)
Comment (chanchiwai-ray) [reply .github/workflows/check.yaml:1]: "Tests" seems to arbitrary, maybe "Code quality check" or "Quality check"

**Review by chanchiwai-ray — DISMISSED** [review]

File: .github/workflows/check.yaml (hunk 2 above)
Comment (rgildein) [reply .github/workflows/check.yaml:40]: Should we use also size? Not sure if for ARM64 we needed such label, but for amd64 we are using it.

Comment (jneo8) [reply .github/workflows/check.yaml:40]: This is the label of github official arm64 runner, we will need it if our decision is to use github official one.

File: .github/workflows/check.yaml (hunk 1 above)
Comment (samuelallan72) [reply .github/workflows/check.yaml:1]: But this is more than quality, this is lints, unit tests, and functional tests.  So tests covers everything.  Also note that with the full path to the job as shown in the github status, it's clear what's running.  The original title is too long and contains redundent information, especially when seen with the job names.

Comment (chanchiwai-ray) [reply .github/workflows/check.yaml:1]: (non-native speaker and non-blocking) but those are should be broadly speaking related to "the quality of the code"

Comment (samuelallan72) [reply .github/workflows/check.yaml:1]: Good point. :+1:   Ok if we have more discussion later after we finish centrally managing them via the terraform repo?  Then we can pick a good name and update it on all the repos to be consistent.

File: .github/workflows/check.yaml (hunk 2 above)
Comment (samuelallan72) [reply .github/workflows/check.yaml:40]: If there are no objections, I'm going to leave it as-is for now.  We can easily change it later and be consistent across projects once it's managed via terraform.  It's working for now. :)

**Review by jneo8 — APPROVED** [review]

**Review by chanchiwai-ray — DISMISSED** [review]

**Review by chanchiwai-ray — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

**Pull request merged**